Docs 8200 edit branch built in dashboards (#132)
mulesoft/docs-monitoring: modules/ROOT/pages/app-dashboards.adoc

[%header,cols="1,4"]
|===
| Built-In Dashboards | Charts
| <<overview-charts>> a|

* Inbound - Total Requests by Response Type
* Inbound - Average Response Time
* Mule Messages
* Outbound - Total Requests by Response Type
* Outbound - Average Response Time
* JVM - CPU % Utilization
* JVM - Heap Used

> * JVM - Thread count

Author: ```suggestion
* JVM - Thread Count
```

| <<inbound-charts>> a|

* Total Inbound Requests by Response Type
* Average Response Time
* Response Time by Endpoint
* Total Requests by Endpoint

> * Response Time graphs (99, 90, 75, and 50 percentile)

Author: Would be lower case graph; this line represents several graphs, so "graph" isn't part of the name for these. Second part applies, however.

> * Response Time graphs (99, 90, 75, and 50 percentile)

Author: ```suggestion
* Response Time graphs (99, 90, 75, and 50 percentiles)
```

* Maximum Response Time
* Total Failed Inbound Requests
* Total Slow Requests by Endpoint
* Inbound - Maximum Response Time
* Inbound Protocol (Titanium only)
* Inbound Endpoint (Titanium only)

[NOTE]
Inbound metrics are retained for 24 hours.

> //* Total Inbound Calls (success or failure)

Author: ```suggestion
```

> //* Total Inbound by Endpoint Outbound

Author: ```suggestion
```

> //* Average Response Time inbound

Author: ```suggestion
```

| <<outbound-charts>> a|

* Total Outbound Requests by Response Type
* Average Response Time
* Average Response Time by HTTP Endpoint
* Total Requests by HTTP Endpoint

> * Response Time graphs (99, 90, 75, and 50 percentile outbound)

Author: ```suggestion
* Response Time graphs (99, 90, 75, and 50 percentiles outbound)
```

JVM - Heap Used::
The average amount of heap memory utilized by the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule JVM process.

JVM - Thread Count::
The average number of threads of the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule JVM process.

[[inbound-charts]]
== Inbound

Inbound metrics are retained for 24 hours.
The following charts appear on the *Inbound* dashboard:

Inbound - Total Requests by Response Type::
The sum of inbound requests, grouped by type of response in automatic intervals. Responses are `OK` or `Failed`. If a Mule app has more than one connector type, all types are included in this metric.

Inbound - Average Response Time::
The average response time of inbound requests, grouped in automatic intervals. If a Mule app has more than one connector type, all types are included in this metric.

Inbound - Response Time by Endpoint::

> The average, maximum, minimum, and total response times, grouped by endpoint.

Author: ```suggestion
The average, maximum, minimum, and total response times, grouped by endpoint. For more information about which endpoints appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

Inbound - Total Slow Requests by Endpoint::
The sum of inbound requests whose response time is greater than 1 second, grouped by endpoint in automatic intervals. The sum of total calls in which the response time is greater than 1000 ms.

The following charts are available only if you are using Unified Agent and have a Titanium subscription to Anypoint Platform:

Inbound - Protocols::
The sum of inbound requests, grouped by protocol in automatic intervals.

Inbound - Endpoints::
The sum of inbound requests, grouped by remote hosts (endpoints) in automatic intervals.

[[outbound-charts]]
== Outbound

The following charts appear on the *Outbound* dashboard:

Outbound - Total Requests by Response Type::
The sum of outbound requests, grouped by type of response in automatic intervals. Responses are `OK` or `Failed`. If a Mule app has more than one connector type, all types are included in this metric.

Outbound - Average Response Time::
The average response time of outbound requests, grouped by application in automatic intervals. If a Mule app has more than one connector type, all types are included in this metric.

Outbound - Response Time by HTTP Endpoint::

> The average, maximum, minimum, and total response times, grouped by endpoint.

Author: ```suggestion
The average, maximum, minimum, and total response times, grouped by endpoint. For more information about which endpoints appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

Inbound Requests by Flow::
The sum of inbound requests, grouped by flow in automatic intervals.

Outbound Requests by Flow::
The sum of outbound requests, grouped by flow in automatic intervals.

Failures by Flow::
The sum of failed inbound requests, grouped by flow in automatic intervals.

Slow Inbound Requests by Flow::
The sum of inbound requests whose response time is greater than 1000 ms, grouped by flow.

Slow Outbound Requests by Flow::
The sum of outbound requests whose response time is greater than 1000 ms, grouped by flow.

[[connectors-charts]]
== Connectors

If you don't see the *Connectors* tab, verify that you have a Titanium subscription to Anypoint Platform and that the app you selected is a hybrid app deployed to a server, server group, or cluster. See xref:monitor-connectors.adoc[Monitor Connectors].

The following charts appear on the *Connectors* dashboard:

Total Connector Requests::
The sum of outbound requests grouped by connector type in automatic intervals.

Response Time by Connector::
The average outbound response time grouped by connector type in automatic intervals.

Salesforce Requests::

> The average number of Salesforce connector requests, grouped by operation in automatic intervals.

Author: ```suggestion
The average number of Anypoint Connector for Salesforce (Salesforce Connector) requests, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

Salesforce Response Time::

> The average response time of Saleforce connector requests, grouped in automatic intervals.

Author: ```suggestion
The average response time of Anypoint Connector for Salesforce (Salesforce Connector) requests, grouped in automatic intervals.
```

Salesforce Failures::

> The average number of Salesforce connector requests failures, grouped by operation in automatic intervals.

Author: ```suggestion
The average number of Anypoint Connector for Salesforce (Salesforce Connector) requests failures, grouped by operation in automatic intervals.
```

Database Requests::
The average number of Database connector requests, grouped by operation in automatic intervals.

Database Response Time::

> The average response time of Database connector requests, grouped in automatic intervals.

Author: ```suggestion
The average response time of Anypoint Connector for Database (Database Connector) requests, grouped in automatic intervals.
```

Database Failures::

> The average number of Database connector requests failures, grouped by operation in automatic intervals.

Author: ```suggestion
The average number of Anypoint Connector for Database (Database Connector) requests failures, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection]
```

* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500ms; red if the response time is greater than or equal to 500 ms.
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600ms; red if the response time is greater than or equal to 600 ms.

Outbound - Total Requests by HTTP Endpoint::
The sum of outbound requests in the selected time period, grouped by endpoint. For more information about which endpoints appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].

Outbound - Response Time (99, 90, 75, and 50 percentiles)::
The average response time of the 99, 90, 75, and 50 percentiles of outbound requests, grouped by application in automatic intervals. The unit for these metrics is milliseconds (ms).

Outbound - Failure Total Request Count::
The sum of failed outbound requests, grouped by type of response in automatic intervals.

Outbound - Ports::
The sum of outbound requests, grouped by remote port.

> 

Author: ```suggestion
```

The following charts appear on the *Connectors* dashboard:

Total Connector Requests::
The sum of outbound requests grouped by connector type in automatic intervals.

Response Time by Connector::
The average outbound response time grouped by connector type in automatic intervals.

Salesforce Requests::
The average number of Anypoint Connector for Salesforce (Salesforce Connector) requests, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].

Salesforce Response Time::
The average response time of Anypoint Connector for Salesforce (Salesforce Connector) requests, grouped in automatic intervals.

Salesforce Failures::
The average number of Anypoint Connector for Salesforce (Salesforce Connector) requests failures, grouped by operation in automatic intervals.

Database Requests::

> The average number of Anypoint Connector for Database (Database Connector) requests, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection]

Author: ```suggestion
The average number of Anypoint Connector for Database (Database Connector) requests, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

Database Response Time::
The average response time of Anypoint Connector for Database (Database Connector) requests, grouped in automatic intervals.

Database Failures::

> The average number of Anypoint Connector for Database (Database Connector) requests failures, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection]

Author: ```suggestion
The average number of Anypoint Connector for Database (Database Connector) requests failures, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

HTTP Requests::
The average number of HTTP connector requests, grouped by operation in automatic intervals.

HTTP Response Time::
The average response time of HTTP connector requests, grouped in automatic intervals.

HTTP Failures::

> The average number of HTTP connector requests failures, grouped by operation in automatic intervals.

Author: ```suggestion
The average number of HTTP connector requests failures, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

The sum of outbound requests, grouped by type of response in automatic intervals. Responses are `OK` or `Failed`. If a Mule app has more than one connector type, all types are included in this metric.

Outbound - Average Response Time::
The average response time of outbound requests, grouped by application in automatic intervals. If a Mule app has more than one connector type, all types are included in this metric.

Outbound - Response Time by HTTP Endpoint::
The average, maximum, minimum, and total response times, grouped by endpoint.
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600ms; red if the response time is greater than or equal to 600 ms.
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500ms; red if the response time is greater than or equal to 500 ms.
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600ms; red if the response time is greater than or equal to 600 ms.

Outbound - Total Requests by HTTP Endpoint::

> The sum of outbound requests in the period, grouped by endpoint.

Author: ```suggestion
The sum of outbound requests in the selected period, grouped by endpoint. For more information about which endpoints appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

Outbound - Response Time (99, 90, 75, and 50 percentiles)::
The average response time of the 99, 90, 75, and 50 percentiles of outbound requests, grouped by application in automatic intervals. The unit for these metrics is milliseconds (ms).

Outbound - Failure Total Request Count::
The sum of failed outbound requests, grouped by type of response in automatic intervals.

Outbound - Ports::
The sum of outbound requests, grouped by remote port in automatic intervals.

Outbound - Operations::

> The sum of outbound requests, grouped by operation in automatic intervals.

Author: ```suggestion
The estimated count of outbound requests, grouped by operation. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

The average, maximum, and minimum response times, grouped by endpoint. Cell color varies according to the following thresholds:
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.

Mule Messages::
The number of Mule messages processed in the runtime, grouped in automatic intervals.

Inbound Response Time graphs (99, 90, 75, and 50 percentiles)::
The response time of the 99th, 90th, 75th, and 50th percentile of inbound requests, grouped by application in automatic intervals.

Outbound - Average Response Time::
The average response time of outbound requests, grouped in automatic intervals.

Outbound - Average Response Time Grouped by HTTP Endpoint::

> The average, maximum, and minimum response times, grouped by endpoint. Cells color varies according to the following thresholds:

Author: ```suggestion
The average, maximum, and minimum response times, grouped by endpoint. For more information about which endpoints appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection]. Cells color varies according to the following thresholds:
```

* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.

Outbound - Response Time graphs (99, 90, 75, and 50 percentiles)::
The response time of the 99th, 90th, 75th, and 50th percentile of outbound requests, grouped in automatic intervals.

[[failures-charts]]
== Failures

The following charts appear on the *Failures* dashboard:

Total Failed Inbound Requests::
The sum of failed inbound requests in the period, grouped in automatic intervals.

Total Failed Outbound Requests::
The sum of failed outbound requests in the period, grouped in automatic intervals.

Inbound - Total Requests by Response Type::
The sum of inbound requests, grouped by type of response and automatic interval.

Outbound - Total Requests by Response Type::
The sum of inbound requests, grouped by type of response and automatic interval.

Inbound - Failure Response Time by Endpoint::

> The average, maximum, and minimum response times of failed inbound requests, grouped by endpoint. Cell color varies according to the following thresholds:

Author: ```suggestion
The average, maximum, and minimum response times of failed inbound requests, grouped by endpoint. For more information about which endpoints appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection]. Cell color varies according to the following thresholds:
```

* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.

Inbound - Total Failed Requests by Endpoint::
The sum of failed inbound requests, grouped by endpoint.

Outbound - Failure Response Time by HTTP endpoint::

> The average, maximum, and minimum response times of failed outbound requests, grouped by endpoint. Cell color varies according to the following thresholds:

Author: ```suggestion
The average, maximum, and minimum response times of failed outbound requests, grouped by endpoint. For more information about which endpoints appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection]. Cell color varies according to the following thresholds:
```

The sum of inbound requests, grouped by type of response in automatic intervals. Responses are `OK` or `Failed`. If a Mule app has more than one connector type, all types are included in this metric.

Inbound - Average Response Time::
The average response time of inbound requests, grouped in automatic intervals. If a Mule app has more than one connector type, all types are included in this metric.

Inbound - Response Time by Endpoint::
The average, maximum, minimum, and total response times, grouped by endpoint.
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600ms; red if the response time is greater than or equal to 600 ms.
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500ms; red if the response time is greater than or equal to 500 ms.
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.

Inbound - Total Requests by Endpoint::
The sum of inbound requests in the period, grouped by endpoint.

> Inbound - Response Time graphs (99, 90, 75, and 50 percentile)::

Author: ```suggestion
Inbound - Response Time graphs (99, 90, 75, and 50 percentiles)::
```

The average number of requests failures of connectors of type `other`, grouped in automatic intervals.

[[performance-charts]]
== Performance

The following charts appear on the *Performance* dashboard:

Inbound - Average Response Time::
The average response time of inbound requests, grouped in automatic intervals.

Inbound - Average Response Time Grouped by Endpoint::
The average, maximum, and minimum response times, grouped by endpoint. Cell color varies according to the following thresholds:
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.

Mule Messages::
The number of Mule messages processed in the runtime, grouped in automatic intervals.

> Inbound Response Time graphs (99, 90, 75, and 50 Percentile)::

Author: ```suggestion
Inbound Response Time graphs (99, 90, 75, and 50 percentiles)::
```

The response time of the 99th, 90th, 75th, and 50th percentile of inbound requests, grouped by application in automatic intervals.

Outbound - Average Response Time::
The average response time of outbound requests, grouped in automatic intervals.

Outbound - Average Response Time Grouped by HTTP Endpoint::
The average, maximum, and minimum response times, grouped by endpoint. Cells color varies according to the following thresholds:
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.

> Outbound Response Time graphs (99, 90, 75, and 50 Percentile)::

Author: ```suggestion
Outbound - Response Time graphs (99, 90, 75, and 50 percentiles)::
```

| <<connectors-charts>> a|

* Connector Message Volume

* Response Time by Connector
* Salesforce Requests
* Salesforce Response Time
* Salesforce Failures
* Database Requests
* Database Response Time
* Database Failures
* HTTP Requests
* HTTP Response Time
* HTTP Failures
* Web Services Requests
* Web Services Response Time
* Web Services Failures
* FTP Requests
* FTP Response Time
* FTP Failures
* SFTP Requests
* SFTP Response Time
* SFTP Failures
* SMTP Requests
* SMTP Response Time
* SMTP Failures
* Object Store Requests
* Object Store Response Time
* Object Store Failures
* Anypoint MQ Operations - Inbound
* Anypoint MQ Response Time - Inbound
* Anypoint MQ Failures - Inbound
* Anypoint MQ Operations - Outbound
* Anypoint MQ Response Time - Outbound
* Anypoint MQ Failures - Outbound
* Other Connector Requests
* Other Connector Response Time
* Other Connector Failures

If you don't see the *Connectors* tab, verify that you have a Titanium subscription to Anypoint Platform and that the app you selected is a hybrid app deployed to a server, server group, or cluster. See xref:monitor-connectors.adoc[Monitor Connectors].

> | <<performance-charts>>* a|

Author: ```suggestion
| <<performance-charts>> a|
```

* Inbound - Average Response Time
* Inbound - Average Response Time Grouped by Endpoint
* Mule Messages

> * Inbound - Response Time graphs (99, 90, 75, and 50 Percentile) - Inbound

Author: ```suggestion
* Inbound - Response Time graphs (99, 90, 75, and 50 percentiles)
```

* Inbound - Response Time by Endpoint
* Outbound - Average Response Time
* Outbound - Average Response Time Grouped by Endpoint
* Outbound - Response Time by HTTP Endpoint

> * Outbound - Response Time graphs (99, 90, 75, and 50 Percentile)

Author: ```suggestion
* Outbound - Response Time graphs (99, 90, 75, and 50 percentiles)
```

| <<failures-charts>> a|

* Total Failed Inbound Requests
* Total Failed Outbound Requests
* Inbound - Total Requests by Response Type
* Outbound - Total Requests by Response Type
* Inbound - Failure Response Time by Endpoint
* Inbound - Total Failed Requests by Endpoint
* Outbound - Failure Response Time by HTTP Endpoint
* Outbound - Total Failed Requests by HTTP Endpoint
| <<jvm-charts>> a|

* Garbage Collection Count

> * Garbage collection Time

Author: ```suggestion
* Garbage Collection Time
```

* ParNew Collection Count
* ParNew collection Time
* Classes (Loaded, Loaded Total, Unloaded)
* Heap (Committed, Used)
* Thread Count
* JVM Uptime
* Par Eden graphs (Used, Max, Init, and Committed)
* Par Survivor graphs (Used, Max, Init, and Committed)

> * Metaspace graphs (Used, Max, Init, Committed)

Author: ```suggestion
* Metaspace graphs (Used, Max, Init, and Committed)
```

> * Code Cache graphs (Used, Max, Init, Committed)

Author: ```suggestion
* Code Cache graphs (Used, Max, Init, and Committed)
```

> * Compressed Class Space graphs (Used, Max, Init, Committed)

Author: ```suggestion
* Compressed Class Space graphs (Used, Max, Init, and Committed)
```

Mule Messages::
The total number of Mule messages that are processed successfully, grouped in automatic intervals.

Outbound - Total Requests by Response Type::
The sum of outbound requests, grouped by type of response in automatic intervals. Responses are OK or Failed. If a Mule app has more than one connector type, all types are included in this metric.

Outbound - Average Response Time::
The average response time of outbound requests, grouped in automatic intervals. If a Mule app has more than one connector type, all types are included in this metric.

JVM - CPU % Utilization::
The average percentage of CPU utilized by the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule runtime JVM process. The calculation for CPU utilization is relative to the physical server on which the Mule runtime is deployed. For example, in RTF, the resource usage here indicates the the JVM's utilization with respect to the entire server, and not with respect to the Pod maximum CPU allocation.

JVM - Heap Used::

> The average amount of heap memory utilized by the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule runtime JVM process.

Author: ```suggestion
The average amount of heap memory utilized by the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule JVM process.
```

JVM - Thread Count::

> The average number of threads of the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule runtime JVM process.

Author: ```suggestion
The average number of threads of the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule JVM process.
```

[[inbound-charts]]
== Inbound

Inbound metrics are retained for 24 hours.
The following charts appear on the *Inbound* dashboard:

Inbound - Total Requests by Response Type::

> The sum of inbound requests, grouped by type of response in automatic intervals. Responses are OK or Failed. If a Mule app has more than one connector type, all types are included in this metric.

Author: ```suggestion
The sum of inbound requests, grouped by type of response in automatic intervals. Responses are `OK` or `Failed`. If a Mule app has more than one connector type, all types are included in this metric.
```

Inbound - Average Response Time::
The average response time of inbound requests, grouped in automatic intervals. If a Mule app has more than one connector type, all types are included in this metric.

Inbound - Response Time by Endpoint::
The average, maximum, and minimum and total response time, grouped by endpoint.

> * Average: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600ms; red if the response time is greater than or equal to 600 ms.
```

> * Maximum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 500ms; red if the response time is greater than or equal to 500ms.

Author: ```suggestion
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500ms; red if the response time is greater than or equal to 500 ms.
```

> * Minimum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

Inbound - Total Requests by Endpoint::

> The sum of inbound requests in the period, grouped by endpoint.

Author: @jiang-wu selected time period?

> Inbound - Response Time graphs (99, 90, 75, and 50 Percentile)::

Author: ```suggestion
Inbound - Response Time graphs (99, 90, 75, and 50 percentile)::
```

The average response time of the 99, 90, 75, and 50 percentiles of inbound requests, grouped in automatic intervals. The unit for these metrics is milliseconds (ms).

Inbound - Maximum Response Time::
The maximum response time, grouped by endpoint in automatic intervals.

Inbound - Total Failed Requests::
The sum of failed inbound requests, grouped by type of response in automatic intervals.

Inbound - Total Slow Requests by Endpoint::
The sum of inbound requests whose response time is greater than 1 second, grouped by endpoint in automatic intervals. The sum of total calls in which the response time is greater than 1000 ms.

The following charts are available only if you are using Unified Agent and have a Titanium subscription to Anypoint Platform:

Inbound - Protocols::
The sum of inbound requests, grouped by protocol in automatic intervals.

Inbound - Endpoints::
The sum of inbound requests, grouped by remote hosts (endpoints) in automatic intervals.

[[outbound-charts]]
== Outbound

The following charts appear on the *Outbound* dashboard:

Outbound - Total Requests by Response Type::

> The sum of outbound requests, grouped by type of response in automatic intervals. Responses are OK or Failed. If a Mule app has more than one connector type, all types are included in this metric.

Author: ```suggestion
The sum of outbound requests, grouped by type of response in automatic intervals. Responses are `OK` or `Failed`. If a Mule app has more than one connector type, all types are included in this metric.
```

Outbound - Average Response Time::
The average response time of outbound requests, grouped by application in automatic intervals. If a Mule app has more than one connector type, all types are included in this metric.

Outbound - Response Time by HTTP Endpoint::

> The average, maximum, and minimum and total response time, grouped by endpoint.

Reviewer: BREAK here to ensure GitHub stability.

> The average, maximum, and minimum and total response time, grouped by endpoint.

Reviewer: RESUME

> * Average: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600ms; red if the response time is greater than or equal to 600 ms.
```

> * Maximum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 500ms; red if the response time is greater than or equal to 500ms.

Author: ```suggestion
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500ms; red if the response time is greater than or equal to 500 ms.
```

> * Minimum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600ms; red if the response time is greater than or equal to 600 ms.
```

Outbound - Total Requests by HTTP Endpoint::
The sum of outbound requests in the period, grouped by endpoint.

> Outbound - Response Time (99, 90, 75, and 50 percentile)::

Reviewer: ```suggestion
Outbound - Response Time (99, 90, 75, and 50 percentiles)::
```

The average response time of the 99, 90, 75, and 50 percentiles of outbound requests, grouped by application in automatic intervals. The unit for these metrics is milliseconds (ms).

Outbound - Failure Total Request Count::
The sum of failed outbound requests, grouped by type of response in automatic intervals.

Outbound - Ports::
The sum of outbound requests, grouped by remote port in automatic intervals.

> // ports?

Reviewer: resolved?

Author: ```suggestion
```

Outbound - Operations::
The sum of outbound requests, grouped by operation in automatic intervals.

Outbound - Protocols::
The sum of outbound requests, grouped by protocol in automatic intervals.

Outbound - Endpoints::
The sum of outbound requests, grouped by remote hosts (endpoints) in automatic intervals.

[[flows-charts]]
== Flows

The following charts appear on the *Flows* dashboard:

Inbound Response Time by Flow::
The average inbound response time, grouped by flow.

Outbound Response Time by Flow::
The average outbound response time, grouped by flow.

Inbound Requests by Flow::
The sum of inbound requests, grouped by flow.

Outbound Requests by Flow::
The sum of outbound requests, grouped by flow.

Failures by Flow::
The sum of failed inbound requests, grouped by flow.

> //tbd?

Reviewer: resolved?

Author: ```suggestion
```

Slow Inbound Requests by Flow::
The sum of inbound requests whose response time is greater than 1000 ms, grouped by flow.

Slow Outbound Requests by Flow::
The sum of outbound requests whose response time is greater than 1000 ms, grouped by flow.

Inbound Response Time by Flow::

> The average, minimum, and maximum inbound response time, grouped by flow.

Reviewer: ```suggestion
The average, minimum, and maximum inbound response times, grouped by flow.
```

Reviewer: GLOBAL
I also just noticed that these definitions are fragments, which do not require periods...unless one or more items in a single list includes a complete sentence, in which case all take periods.

Outbound Response Time by Flow::
The average, minimum, and maximum outbound response time, grouped by flow.

Inbound Requests by Flow::
The sum of inbound requests, grouped by flow.

Outbound Requests by Flow::
The sum of outbound requests, grouped by flow.

Failures by Flow::
The sum of inbound failed requests, grouped by flow.

> //tbd?

Reviewer: resolved?

Author: ```suggestion
```

Slow Inbound Requests by Flow::
The sum of inbound requests whose response time is greater than 1000 ms, grouped by flow.

Slow Outbound Requests by Flow::
The sum of outbound requests whose response time is greater than 1000 ms, grouped by flow.

[[connectors-charts]]
== Connectors

If you don't see the *Connectors* tab, verify that you have a Titanium subscription to Anypoint Platform and that the app you selected is a hybrid app deployed to a server, server group, or cluster. See xref:monitor-connectors.adoc[Monitor Connectors].

The following charts appear on the *Connectors* dashboard:

Total Connector Requests::
The sum of outbound requests grouped by connector type in automatic intervals.

Response Time by Connector::
The average outbound response time grouped by connector type in automatic intervals.

Salesforce Requests::

> The average number of Salesforce connector requests, grouped by operation in automatic intervals.

Reviewer: What do you mean by "Salesforce connector"? Anypoint Connectors are Anypoint-Platform specific (MuleSoft), although there is a named "Anypoint Connector for Salesforce (Salesforce Connector)."
GLOBAL

Author: @jiang-wu @michael-lee-mulesoft Connector names?

Salesforce Response Time::
The average response time of Saleforce connector requests, grouped in automatic intervals.

Salesforce Failures::
The average number of Salesforce connector requests failures, grouped by operation in automatic intervals.

Database Requests::

> The average number of Database connector requests, grouped by operation in automatic intervals.

Reviewer: ditto: do we mean "Anypoint Connector for Database (Database Connector)"?

Author: ```suggestion
The average number of Anypoint Connector for Database (Database Connector) requests, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection]
```

Database Response Time::
The average response time of Database connector requests, grouped in automatic intervals.

Database Failures::
The average number of Database connector requests failures, grouped by operation in automatic intervals.

HTTP Requests::

> The average number of HTTP connector requests, grouped by operation in automatic intervals.

Reviewer: ditto; GLOBAL in this topic

Author: ```suggestion
The average number of HTTP connector requests, grouped by operation in automatic intervals. For more information about which operations appear, see the section about xref:monitoring::app-dashboards.adoc#endpoint-operation-metrics[endpoint and operation data collection].
```

Object Store Requests::
The average response time of Object Store connector requests, grouped by operation and automatic intervals.

Object Store Response Time::
The average number of Object Store connector request failures, grouped by operation and automatic intervals.

Object Store Failures::
The average number of Anypoint MQ connector inbound requests, grouped by operation and automatic intervals.

Anypoint MQ Operations - Inbound::
The average response time of Anypoint MQ connector inbound requests, grouped by operation and automatic intervals.

Anypoint MQ Response Time - Inbound::
The average number of Anypoint MQ connector inbound request failures, grouped by operation and automatic intervals.

Anypoint MQ Failures - Inbound::
The average number of Anypoint MQ connector outbound requests, grouped by operation and automatic intervals.

Anypoint MQ Operations - Outbound::
The average response time of Anypoint MQ connector outbound requests, grouped by operation and automatic intervals.

Anypoint MQ Response Time - Outbound::
The average number of Anypoint MQ connector outbound request failures, grouped by operation and automatic intervals.

Anypoint MQ Failures - Outbound::

> The average number of requests of connectors of type other, grouped by operation in automatic intervals.

Author: ```suggestion
The average number of requests of connectors of type `other`, grouped by operation in automatic intervals.
```

Other Connector Requests::

> The average number of requests of connectors of type other, grouped in automatic intervals.

Reviewer: "of type other" = "of type Other"? "of type `other` (or 'Other`)?
GLOBAL

Author: ```suggestion
The average number of requests of connectors of type `other`, grouped in automatic intervals.
```

Other Connector Response Time::

> The average response time of requests for connectors of type other, grouped in automatic intervals.

Author: ```suggestion
The average response time of requests for connectors of type `other`, grouped in automatic intervals.
```

Other Connector Failures::

> The average number of requests failures of connectors of type other, grouped in automatic intervals.

Author: ```suggestion
The average number of requests failures of connectors of type `other`, grouped in automatic intervals.
```

[[performance-charts]]
== Performance

The following charts appear on the *Performance* dashboard:

Inbound - Average Response Time::
The average response time of inbound requests, grouped in automatic intervals.

Inbound - Average Response Time Grouped by Endpoint::

> The average, maximum, and minimum response times, grouped by endpoint. Cells will vary in color according to the following thresholds:

Reviewer: Avoid future tense:
```suggestion
The average, maximum, and minimum response times, grouped by endpoint. Cell color varies according to the following thresholds:
```

> * Average: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

> * Maximum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 500ms; red if the response time is greater than or equal to 500ms.

Author: ```suggestion
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
```

> * Minimum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

Mule Messages::

> The number of Mule Messages processed in the runtime, grouped in automatic intervals.

Reviewer: ```suggestion
The number of Mule messages processed in the runtime, grouped in automatic intervals.
```

Inbound Response Time graphs (99, 90, 75, and 50 Percentile)::

> The response time of the 99th, 90th, 75th, and 50th percentile of inbound requests, grouped by application in automatic intervals.

Reviewer: Notice that presentation changes here; please standardize percentile language GLOBALLY.

Outbound - Average Response Time::
The average response time of outbound requests, grouped in automatic intervals.

Outbound - Average Response Time Grouped by HTTP Endpoint::

> The average, maximum, and minimum response times, grouped by endpoint. Cells will vary in color according to the following thresholds:

Reviewer: ditto prev. edits, same text earlier

> The average, maximum, and minimum response times, grouped by endpoint. Cells will vary in color according to the following thresholds:

Author: ```suggestion
The average, maximum, and minimum response times, grouped by endpoint. Cells colors vary according to the following thresholds:
```

> * Average: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

> * Maximum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 500ms; red if the response time is greater than or equal to 500ms.

Author: ```suggestion
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
```

> * Minimum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

Outbound Response Time graphs (99, 90, 75, and 50 Percentile)::
The response time of the 99th, 90th, 75th, and 50th percentile of outbound requests, grouped in automatic intervals.

[[failures-charts]]
== Failures

The following charts appear on the *Failures* dashboard:

Total Failed Inbound Requests::
The sum of failed inbound requests in the period, grouped in automatic intervals.

Total Failed Outbound Requests::
The sum of failed outbound requests in the period, grouped in automatic intervals.

Inbound - Total Requests by Response Type::
The sum of inbound requests, grouped by type of response and automatic interval.

Outbound - Total Requests by Response Type::

> The sum of inbound requests, grouped by type of response and automatic interval.

Reviewer: "...and automatic interval" makes more sense to me than "in automatic intervals." However, I still think we should clarify what we mean by that: what are the intervals and who/what determines them"

Author: @jiang-wu What exactly does automatic intervals refer to here, and how can I standardize this?

Inbound - Failure Response Time by Endpoint::

> The average, maximum, and minimum response times of failed inbound requests, grouped by endpoint. Cells will vary in color according to the following thresholds:

Reviewer: ditto GLOBAL edits

Author: ```suggestion
The average, maximum, and minimum response times of failed inbound requests, grouped by endpoint. Cell color varies according to the following thresholds:
```

> * Average: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

> * Maximum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 500ms; red if the response time is greater than or equal to 500ms.

Author: ```suggestion
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
```

> * Minimum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

Inbound - Total Failed Requests by Endpoint::
The sum of failed inbound requests, grouped by endpoint.

Outbound - Failure Response Time by HTTP endpoint::

> The average, maximum, and minimum response times of failed outbound requests, grouped by endpoint. Cells will vary in color according to the following thresholds:

Author: ```suggestion
The average, maximum, and minimum response times of failed outbound requests, grouped by endpoint. Cell color varies according to the following thresholds:
```

> * Average: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Average: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

> * Maximum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 500ms; red if the response time is greater than or equal to 500ms.

Author: ```suggestion
* Maximum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 500 ms; red if the response time is greater than or equal to 500 ms.
```

> * Minimum: Green if the response time is less than 300ms; yellow if the response time is greater than 300ms and less than 600ms; red if the response time is greater than or equal to 600ms.

Author: ```suggestion
* Minimum: Green if the response time is less than 300 ms; yellow if the response time is greater than 300 ms and less than 600 ms; red if the response time is greater than or equal to 600 ms.
```

Outbound - Total Failed Requests by HTTP Endpoint::
The sum of failed outbound requests, grouped by endpoint.

[[jvm-charts]]

> == JVM

Reviewer: I spent some time looking up space types in JVM. I found five memory space types, but two types of heap memory: eden and survivor. I never saw "par", although I did see "per." Would you please advise me on the proper terms and capitalization for these, for the Term Presentation List?

Author: @jiang-wu are we sure this is correct?

The memory metrics for JVM Par Eden space, grouped by worker in automatic intervals.

Par Survivor graphs (Used, Max, Init, and Committed)::
The memory metrics for JVM Par Survivor space, grouped by worker in automatic intervals.

Metaspace graphs (Used, Max, Init, Committed)::
The memory metrics for JVM metaspace, grouped by worker in automatic intervals.

Code Cache graphs (Used, Max, Init, Committed)::
The memory metrics for JVM code cache, grouped by worker in automatic intervals.

Compressed Class Space graphs (Used, Max, Init, Committed)::
The memory metrics for JVM compressed class, grouped by worker in automatic intervals.

Committed Virtual Memory::
The amount of virtual memory guaranteed to be available to the running process, grouped by worker in automatic intervals.

[[infrastructure-charts]]
== Infrastructure

The following charts appear on the *Infrastructure* dashboard:

Application Process CPU Load::
The amount of CPU load the JVM uses.

JVM CPU Time::
The cumulative CPU time consumed by the JVM.

Operating System Swap Space::
The amount of memory the system is swapping, grouped by worker in automatic intervals.

JVM - CPU % Utilization::

> The average percentage of CPU utilized by the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule runtime JVM process. The calculation for CPU utilization is relative to the physical server on which the Mule runtime is deployed. For example, in RTF, the resource usage here indicates the JVM's utilization with respect to the entire server and not with respect to the Pod maximum CPU allocation.

Reviewer: ditto all prev edits

Author: ```suggestion
The average percentage of CPU utilized by the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule JVM process. The calculation for CPU utilization is relative to the physical server on which the Mule runtime is deployed. For example, in RTF, the resource usage here indicates the JVM's utilization with respect to the entire server and not with respect to the pod maximum CPU allocation.
```

JVM - Heap Used::
The average amount of heap memory utilized by the JVM, grouped by worker in automatic intervals. The JVM used in this chart is the Mule runtime JVM process.

Total System Processors::
The number of CPU processors available to the Java virtual machine for each worker.

Total System Memory::
The amount of system memory available for workers over time.

JVM Thread Count - Server::
The average number of threads of the JVM, grouped by worker in automatic intervals.

.Example: Infrastructure Dashboard
image::dashboard-built-in-infrastructure.png[Example: Infrastructure Dashboard]

[[custom-metrics-charts]]
== Custom Metrics

The *Custom metrics* tab displays only if you have a Titanium subscription and select an app running on a supported version of Mule. See xref:anypoint-custom-metrics-connector.adoc[Anypoint Custom Metrics Connector].

[[application-network]]
== Application Network

> The chart on *Application Network* tab shows the xref:visualizer::troubleshooting-visualization.adoc[Troubleshooting visualization] for the selected app, which must be started. The chart shows only direct callers and callees for the selected app. The connections are based on the selected time range, with a max of the most recent seven days.

Author: ```suggestion
The chart on the *Application Network* tab shows the xref:visualizer::troubleshooting-visualization.adoc[Troubleshooting visualization] for the selected app, which must be started. The chart shows only direct callers and callees for the selected app. The connections are based on the selected time range, with a max of the most recent seven days.
```
